Libraries/Utilities/Platform.ios.js

```diff
@@ -67,6 +68,10 @@ const Platform = {
     }
     return false;
   },
+  // $FlowFixMe[unsafe-getters-setters]
+  get isCatalyst(): boolean {
```

Comment:
I'd prefer this to called `isMacCatalyst()` as it's more clear and more inline with how Apple [markets](https://developer.apple.com/mac-catalyst/) the technology.

My general concern is that this will force all platforms to implement a `isMacCatalyst()` method, which is very specific, so I wonder if we should instead go with a method called `isDesktop()` to be more inclusive for our other platforms. That way `react-native-macos` and `react-native-windows` can return `true` for this method as well.

Thoughts @alloy?

Reply:
I like that, I’m just a little concerned about the boundaries. Would this return `true` on iPad OS or what is the definition of a desktop machine?

Reply:
No, the purpose is it would return `true` for macOS (AppKit or UIKit) and Windows. 😄

Although lots of people iPad as a desktop, I think the general expectation is that it is not classified as a desktop. If that's a controversial opinion, then we can consider another name. We already have `isPad()` and `isTV()` methods that describe the *type* of device rather than particular OS, so I think `isDesktop()`  would fit nicely in there.

Reply:
Yeah, perhaps desktops and iPads together are ‘workstations’? In any case, seems like we have enough ways to narrow down the type, so I’m good with ‘desktop’ 👍

Reply:
why not just include it inside Platform.OS

Reply:
@EhsanSarshar because Mac Catalyst isn't a platform itself, it's running the iOS app on macOS, so its behavior should mostly follow iOS except for some cases where we want it to be desktop.

Also, `Platform.OS` is handled by the packager ([Metro](https://github.com/facebook/metro)) during build time, not here in React Native. See this PR for the implementation of `Platform.select` with `mac`: https://github.com/facebook/metro/pull/495.

Reply:
@charpeni Will you update this to be `isDesktop`?

Reply:
Will do!